File: ios/sources/commands/ARESStrokeStyleCommand.m
//
//  ARESStrokeStyleCommand.m
//  ares
//
//  Created by PonyCui on 2018/11/29.
//  Copyright © 2018年 XT Studio. All rights reserved.
//

#import "ARESStrokeStyleCommand.h"
#import "UIColor+ARESParser.h"
#import "ARESView.h"

@implementation ARESStrokeStyleCommand

- (instancetype)initWithValue:(NSString *)value
{
    self = [super init];
    if (self) {
        _color = [UIColor ares_parseColorWithValue:value];
    }
    return self;
}

- (void)draw:(CGContextRef)ctx view:(ARESView *)view {
    CGContextSetStrokeColorWithColor(ctx, self.color.CGColor);
    view.currentState.strokeColor = self.color;
}

@end

File: android/ares/src/main/java/com/xt/ares/commands/ARESStrokeStyleCommand.kt
package com.xt.ares.commands

import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import com.xt.ares.ARESCommand
import com.xt.ares.ARESView
import com.xt.ares.common.ARESColorUtils

class ARESStrokeStyleCommand(value: String): ARESCommand() {

    var color: Int = 0
        private set

    init {
        this.color = ARESColorUtils.parseColorWithValue(value)
    }

    override fun draw(view: ARESView, canvas: Canvas) {
        super.draw(view, canvas)
        view.currentPaint.strokeColor = this.color
    }

}